src/transformers/models/propainter/modeling_propainter.py

```diff
@@ -0,0 +1,4620 @@
+                    (coords1 - coords0),
+                    size=new_size,
+                    mode="bilinear",
+                    align_corners=True,
+                )
+            else:
+                flow_up = self.upsample_flow(coords1 - coords0, up_mask)
+
+        return coords1 - coords0, flow_up
+
+    def forward(self, ground_truth_local_frames, iters=20):
+        batch_size, temporal_length, num_channels, height, width = ground_truth_local_frames.size()
+
+        ground_truth_local_frames_1 = ground_truth_local_frames[:, :-1, :, :, :].reshape(
+            -1, num_channels, height, width
+        )
+        ground_truth_local_frames_2 = ground_truth_local_frames[:, 1:, :, :, :].reshape(
+            -1, num_channels, height, width
+        )
+        _, ground_truth_flows_forward = self._forward(ground_truth_local_frames_1, ground_truth_local_frames_2, iters)
+        _, ground_truth_flows_backward = self._forward(ground_truth_local_frames_2, ground_truth_local_frames_1, iters)
+
+        ground_truth_flows_forward = ground_truth_flows_forward.view(batch_size, temporal_length - 1, 2, height, width)
+        ground_truth_flows_backward = ground_truth_flows_backward.view(
+            batch_size, temporal_length - 1, 2, height, width
+        )
+
+        return ground_truth_flows_forward, ground_truth_flows_backward
+
+
+class ProPainterP3DBlock(nn.Module):
+    def __init__(
+        self,
+        config: ProPainterConfig,
+        in_channels: int,
+        out_channels: int,
+        stride: int,
+        use_residual: bool = False,
+        bias=True,
+    ):
+        super().__init__()
+        self.config = config
+        self.conv1 = nn.Sequential(
+            nn.Conv3d(
+                in_channels,
+                out_channels,
+                kernel_size=(1, config.patch_size, config.patch_size),
+                stride=(1, stride, stride),
+                padding=(0, config.padding, config.padding),
+                bias=bias,
+            ),
+            nn.LeakyReLU(config.negative_slope_default, inplace=True),
+        )
+        self.conv2 = nn.Sequential(
+            nn.Conv3d(
+                out_channels,
+                out_channels,
+                kernel_size=(3, 1, 1),
+                stride=(1, 1, 1),
+                padding=(2, 0, 0),
+                dilation=(2, 1, 1),
+                bias=bias,
+            )
+        )
+        self.use_residual = use_residual
+
+    def forward(self, hidden_state):
+        features1 = self.conv1(hidden_state)
+        features2 = self.conv2(features1)
+        if self.use_residual:
+            hidden_state = hidden_state + features2
+        else:
+            hidden_state = features2
+        return hidden_state
+
+
+class ProPainterEdgeDetection(nn.Module):
+    def __init__(
+        self,
+        config: ProPainterConfig,
+        in_channel: int = 2,
+        out_channel: int = 1,
+        intermediate_channel: int = 16,
+    ):
```

Comment:
same remark, we can move these values to the config and simply initialize the module with the configuration passed, with explicitly named keys